websites/T/TV Time/presence.ts

```diff
@@ -3,73 +3,191 @@ const presence = new Presence({
+						presenceData.details = "Viewing shows filtered by";
+						presenceData.state = `${hrefSplit
+							.at(0)
+							.toUpperCase()}${hrefSplit.slice(1)}`;
+						break;
+					}
+					default: {
+						presenceData.details = "Viewing";
+						presenceData.state = title;
+						presenceData.largeImageKey =
+							document
+								.querySelector('img[src*="/poster/"]')
+								?.getAttribute("src") ??
+							document
+								.querySelector('[property="og:image"]')
+								?.getAttribute("content") ??
+							Assets.Logo;
+						if (title.match(/S[0-9]*E[0-9]/gm)) {
+							presenceData.buttons = [
+								{
+									label: "View Episode",
+									url: href,
+								},
+							];
+						} else {
+							presenceData.buttons = [
+								{
+									label: "View Show",
+									url: href,
+								},
+							];
+						}
+						break;
+					}
+				}
+				break;
+			}
+			case "actor": {
+				presenceData.details = "Viewing";
+				presenceData.state = title;
+				presenceData.largeImageKey =
+					document
+						.querySelector('[property="og:image"]')
+						?.getAttribute("content") ?? Assets.Logo;
+				presenceData.buttons = [
+					{
+						label: "View Actor",
+						url: href,
+					},
+				];
+				break;
+			}
+			case "upcoming": {
+				presenceData.details = "Viewing upcoming shows";
+				presenceData.buttons = [
+					{
+						label: "View Upcoming Shows",
+						url: href,
+					},
+				];
+				break;
+			}
+			case "explore": {
+				if (!document.querySelector('[id="shows-results"]')) {
+					presenceData.details = "Exploring Shows";
+					presenceData.buttons = [
+						{
+							label: "Explore Shows",
+							url: href,
+						},
+					];
+				} else {
+					presenceData.details = "Viewing search results for";
+					presenceData.state = document
+						.querySelector('[id="shows-results"]')
+						.textContent.split('"')[1];
+				}
+				break;
+			}
+			case "user": {
+				switch (splitPathname[4]) {
+					case "profile": {
+						presenceData.details = `Viewing ${title}'s ${
+							document
+								.querySelector('[class="profile-nav"]')
+								.querySelector('[class*="active"]')?.textContent ?? "Profile"
+						}`;
+						presenceData.largeImageKey =
+							document
+								.querySelector('[class="avatar"]')
+								?.firstElementChild?.firstElementChild?.getAttribute("src") ??
+							document
+								.querySelector('[class="avatar"]')
+								?.firstElementChild?.getAttribute("src");
+						break;
+					}
+					case "calendar": {
+						presenceData.details = "Viewing Calendar for";
+						presenceData.state = `${
+							document.querySelector('[class="month-label"]').textContent
+						} ${
+							document.querySelector('[class="day-number"]').textContent
+						} (${document
+							.querySelector('[class="relative-date"]')
+							.textContent.trim()})`;
+						presenceData.buttons = [
+							{
+								label: "View Their Calendar",
+								url: href,
+							},
+						];
```

Comment:
Can you give me an example of viewing someone else's calendar? Seems too privacy invasive

Reply:
> Can you give me an example of viewing someone else's calendar? Seems too privacy invasive

![image](https://user-images.githubusercontent.com/42322979/186117327-2286a730-a74a-4193-a68d-600a4caf74f3.png)

Reply:
```suggestion
						presenceData.buttons = [
							{
								label: "View Calendar",
								url: href,
							},
						];
```